sprocs/grading_jobs_insert_internal.sql

```diff
@@ -83,20 +83,32 @@ BEGIN
 
     INSERT INTO grading_jobs AS gj
         (submission_id,     score,     v2_score, correct,     feedback,
-            partial_scores, auth_user_id,  grading_method)
+            partial_scores, auth_user_id,  grading_method, gradable)
     VALUES
         (submission_id, new_score, new_v2_score, new_correct, new_feedback,
-        new_partial_scores, authn_user_id, grading_method)
+        new_partial_scores, authn_user_id, grading_method, new_gradable)
     RETURNING gj.*
     INTO grading_job;
 
-    -- ######################################################################
-    -- update all parent objects
+    IF new_gradable = FALSE THEN
+        -- ######################################################################
+        -- If the submission is ungradable then we shouldn't update grades
+        -- and use up an attempt
```

Comment:
Should we be setting `submission.gradable = false` here? And also setting `submission.format_errors` (and possibly `submission.feedback`?)

Reply:
Ah yup, good catch :sweat_smile:

Reply:
Oh, no actually this will be already set in the `UPDATE` on lines 56-70

Reply:
Oh right!